### DISCUSSION TITLE
Convince me: Migrating from (headless) Shopify to Swell.

### DISCUSSION THREAD

**Discussion by DennisKraaijeveld** [swellstores/community#125] [Help] [answered]
Hi!

We have two separate E-Commerce brands with both their own storefront, but handled by the same Shopify backend. The custom storefronts are built with Gatsby but like many, we are not happy with Gatsby anymore and we do want to improve performance and functionalities on our store. 

At first I was thinking about Remix/Hydrogen/Storefront API, but I am not sure if I should do that since I had some issues and things missing from their ecosystem. Like selling in bulk, easier to handle multiple markets, automatic return management and flexibility.

But it is a  really tough choice to make, since Shopify is developing in such a rapid tempo, that it isn't easy to let go. So why should I go for Swell? What benefits will I get compared to Shopify? 

We have a total of 400-450k revenue each year, so we are too small for Shopify Plus / Swell Enterprise. But the new store should have multiple markets, return management within the storefront, selling in bulk, bulk discount, B2B and B2C, better filtering options.

**Comment by papertokyo** [comment, marked as answer]
Based on your requirements, it sounds like Swell is a strong fit for your use case. Shopify is doing some really interesting things in terms of tooling, but you'll likely still find the APIs restrictive in comparison. 

Whether that's a dealbreaker or not will come down to the nuances of your implementation needs. My suggestion is to build a small PoC with Next.js doing the things you've found challenging with Shopify and see if you prefer how Swell works. 

You can use our [Horizon storefront](https://github.com/swellstores/horizon) as a reference for many of the patterns you'll need.

**Reply by DennisKraaijeveld** [reply]
> Based on your requirements, it sounds like Swell is a strong fit for your use case. Shopify is doing some really interesting things in terms of tooling, but you'll likely still find the APIs restrictive in comparison.
> 
> Whether that's a dealbreaker or not will come down to the nuances of your implementation needs. My suggestion is to build a small PoC with Next.js doing the things you've found challenging with Shopify and see if you prefer how Swell works.
> 
> You can use our [Horizon storefront](https://github.com/swellstores/horizon) as a reference for many of the patterns you'll need.

Thanks for your reply. I will definitely start on a PoC this week. Some small questions though:

I know we will need to pay 299 for every store we will setup in Swell, but I also know we can power multiple storefronts with one Swell store. So lets say we choose for the latter, could each storefront have its own custom checkout url? Or does it work like Shopify right now where you have:

StorefrontA.com -> StorefrontA.my-shopify.com/checkout
StorefrontB.com -> StorefrontA.my-shopify.com/checkout

**Reply by swellmike** [reply]
@DennisKraaijeveld It's currently one hosted checkout per client/store, no matter how many storefronts you add.

**Reply by DennisKraaijeveld** [reply]
@swellmike Ok clear! So for two custom storefront with one Swell account we can use 2 custom checkout urls?

**Reply by swellmike** [reply]
@DennisKraaijeveld You would need a second Swell account (client/store) if you wanted to manage two different checkouts.
